rootex/framework/components/game/spring_arm_component.cpp

```diff
@@ -0,0 +1,38 @@
+#include "spring_arm_component.h"
+
+DEFINE_COMPONENT(SpringArmComponent);
+
+SpringArmComponent::SpringArmComponent(Entity& owner, const JSON::json& data)
+    : Component(owner)
+    , m_DependencyOnTransformComponent(this)
+{
+}
+
+JSON::json SpringArmComponent::getJSON() const
+{
+	return JSON::json::object();
+}
+
+bool SpringArmComponent::setupData()
+{
+	m_DesiredLocalPosition = m_DependencyOnTransformComponent.getComponent()->getPosition();
+	return true;
+}
+
+void SpringArmComponent::draw()
+{
+	TransformComponent* transform = this->getOwner().getComponent<TransformComponent>();
```

Comment:
This is a non-idiomatic way of getting components. Use `getTransformComponent()` directly. Also, in the future, usually try to stay away from doing something like `this->` just because it's been our convention